examples/step-67/doc/intro.dox

```diff
@@ -0,0 +1,680 @@
+the same as before, all handled transparently by the evalautor. We also note
+that the variant with a single evaluator chosen in the code below is not the
+only one -- we could also have used separate evalators for the separate
+components $\rho$, $\rho \mathbf u$, and $E$; given that we treat all
+components similarly (also reflected in the way we state the equation as a
+vector system), this would be more complicated here. As before, the
+FEEvaluation class provides explicit vectorization by combining the operations
+on several cells (and faces), involving data types called
+VectorizedArray. Since the arithmetic operations are overloaded for this type,
+we do not have to bother with it all that much, except for the evaluation of
+functions through the Function interface, where we need to provide particular
+<i>vectorized</i> evaluations for several positions at once.
+
+A more substantial change in this program is the operation at quadrature
+points: Here, the multi-component evaluators provide us with return types not
+discussed before. Whereas FEEvaluation::get_value() would return a scalar
+(more precisely, a VectorizedArray type due to vectorization across cells) for
+the Laplacian of step-37, it now returns a type that is
+`Tensor<1,dim+2,VectorizedArray<Number>>`. Likewise, the gradient type is now
+`Tensor<1,dim+2,Tensor<1,dim,VectorizedArray<Number>>>`, where the outer
+tensor collects the `dim+2` components of the Euler system, and the the inner
+tensor the partial derivatives in the various directions. For example, the
+flux of the Euler system is of this type. In order to reduce the amount of
+code we have to write for spelling out these types, we use the C++ `auto`
+keyword where possible.
+
+From an implementation point of view, the nonlinearity is not a big
+difficulty: It is introduced naturally as we express the terms of the Euler
+weak form, for example in the form of the momentum term $\rho \mathbf{u}
+\otimes \mathbf{u}$. To obtain this expression, we first deduce the velocity
+$\mathbf{u}$ from the momentum variable $\rho \mathbf{u}$. Given that $\rho
+\mathbf{u}$ is represented as a $p$-degree polynomial, as is $\rho$, the
+velocity $\mathbf{u}$ is a rational expression in terms of the reference
+coordinates $\hat{\mathbf{x}}$. As we perform the multiplication $(\rho
+\mathbf{u})\otimes \mathbf{u}$, we obtain an expression that contains a
+rational expression of two polynomials, with polynomial degree $2p$ in the
+numerator and polynomial degree $p$ in the denominator. Combined with the
+gradient of the test function, the integrand is of degree $3p$ in the
+numerator and $p$ in the denominator already for affine (undeformed) cell
+geometries. For curved cells, additional polynomial and rational expression
+appear. At this point, one usually needs to give up on insisting on exact
+integration, and take whatever accuracy the Gaussian (more precisely,
+Gauss--Legrende) quadrature provides. As that formulas only integrate
```

Comment:
I'll note that that is the same when you integrate for the Laplace equation: The term `grad phi_i . grad phi_j` on non-affine cells is a rational function because the gradient involves the inverse of the Jacobian matrix; this means dividing by the determinant of the Jacobian, which is a polynomial of degree `p*d`. In other words, for non-affine cells, we don't even integrate exactly for the Laplace equation, even though it is linear.

Reply:
```suggestion
Gauss--Legrende) quadrature provides. As these formulas only integrate
```

Reply:
I added a short comment on the inexact integration of the Laplace equation on non-affine cells.